### PR TITLE
feat: add `get` method to store

### PR DESCRIPTION
**Pull request by gamemaker1**
Uses a script to fetch the value and the expiry time for a key.

Note that this pr involves quite a bit of refactoring too, if you want, I could separate that into a different pr.

Fixes https://github.com/express-rate-limit/express-rate-limit/issues/404.

### CODE REVIEW

**Review by nfriedly — APPROVED** [review]
Looks good! It was a bit of work to follow all the changes, but I think I like the final result.

File: source/lib.ts [hunk 1]
```diff
@@ -4,9 +4,43 @@
...
+ * Note that the responses returned by the `get` and `increment` scripts are
+ * the same, so this function can be used with both.
+ */
+const parseScriptResponse = (results: RedisReply): ClientRateLimitInfo => {
+	if (!Array.isArray(results))
+		throw new TypeError('Expected result to be array of values')
+	if (results.length !== 2)
+		throw new Error(`Expected 2 replies, got ${results.length}`)
+
+	const totalHits = toInt(results[0] === false ? 0 : results[0])
```
[the remark this answers is not in the document]
Comment (gamemaker1) [reply source/lib.ts]: Good catch, thanks!

[the remark this answers is not in the document]
Comment (gamemaker1) [reply source/lib.ts]: I misunderstood the suggested change and created a [bug instead](https://github.com/express-rate-limit/express-rate-limit/issues/404#issuecomment-1748386293) :(

[the remark this answers is not in the document]
Comment (gamemaker1) [reply source/lib.ts]: Fixed in v4.1.1

**Pull request merged**